charm.go

```diff
@@ -93,3 +95,27 @@ func IsUnsupportedSeriesError(err error) bool {
 	_, ok := err.(*unsupportedSeriesError)
 	return ok
 }
+
+// isMinVersionError reports whether the given error indicates a minmum version
+// too high for this version of juju.
+func IsMinVersionError(err error) bool {
+	_, ok := err.(minJujuVersionErr)
+	return ok
+}
+
+// CheckMinVersion reports an error that reports true from IsMinVersionErr if
+// the given charm has a minimum juju version specified that is higher than the
+// given juju version number.
+func CheckMinVersion(ch Charm, jujuVersion version.Number) error {
```

Comment:
this doesn't quite feel like the right place for the error -- and the reference to an "environment" below makes me doubly sure (charm shouldn't need to reference the concept at all). Please move this stuff inside juju/juju